boring/src/ssl/mod.rs

```diff
@@ -1180,6 +1227,50 @@ impl SslContextBuilder {
         }
     }
 
+    /// Configures a custom session ticket key callback for session resumption.
+    ///
+    /// Session Resumption uses the security context (aka. session tickets) of a previous
+    /// connection to establish a new connection via an abbreviated handshake. Skipping portions of
+    /// a handshake can potentially yield performance gains.
+    ///
+    /// An attacker that compromises a server's session ticket key can impersonate the server and,
+    /// prior to TLS 1.3, retroactively decrypt all application traffic from sessions using that
+    /// ticket key. Thus ticket keys must be regularly rotated for forward secrecy.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if this `Ssl` is associated with a RPK context.
+    ///
+    /// # Safety
+    ///
+    /// The application is responsible for correctly setting the key_name, iv, encryption context
+    /// and hmac context. See the [`SSL_CTX_set_tlsext_ticket_key_cb`] docs for additional info.
+    ///
+    /// [`SSL_CTX_set_tlsext_ticket_key_cb`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_CTX_set_tlsext_ticket_key_cb
+    #[corresponds(SSL_CTX_set_tlsext_ticket_key_cb)]
+    pub unsafe fn set_ticket_key_callback_unsafe<F>(&mut self, callback: F)
+    where
+        F: Fn(
+                &SslRef,
+                &mut [u8; 16],
+                &mut [u8; ffi::EVP_MAX_IV_LENGTH as usize],
+                &mut MaybeUninit<ffi::EVP_CIPHER_CTX>,
+                &mut MaybeUninit<ffi::HMAC_CTX>,
```

Comment:
> In both modes, ctx and hmac_ctx will already have been initialized with [EVP_CIPHER_CTX_init](https://commondatastorage.googleapis.com/chromium-boringssl-docs/cipher.h.html#EVP_CIPHER_CTX_init) and [HMAC_CTX_init](https://commondatastorage.googleapis.com/chromium-boringssl-docs/hmac.h.html#HMAC_CTX_init), respectively. callback configures hmac_ctx with an HMAC digest and key, and configures ctx for encryption or decryption, based on the mode.
>
> -- [Source](https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_CTX_set_tlsext_ticket_key_cb)

Based on the docs these should be initialized by Boring. The callback is only responsible for configuring the hmac/cipher_contexts not initializing them so I feel the `MaybeUninit` is going to lead to confusion.